docs/installation/server-config/config-reference.rst

```diff
@@ -0,0 +1,221 @@
+
+This number of stream threads in an instance of the Kafka Streams application. The stream processing code runs in these
+threads. For more information about Kafka Streams threading model, see :ref:`streams_architecture_threads`.
+
+
+KSQL Query Settings
+------------------
+
+These configurations control how KSQL executes queries. These configurations can be specified via the ``ksql-server.properties``
+file or via ``SET`` in a KSQL CLI. For example, ``ksql.service.id`` and ``ksql.persistent.prefix``.
+
+
+.. _ksql-fail-on-deserialization-error:
+
+----------------------------------
+ksql.fail.on.deserialization.error
+----------------------------------
+
+Indicates whether to fail if corrupt messages are read. KSQL decodes messages at runtime when reading from a Kafka topic. The
+decoding that KSQL uses depends on what's defined in STREAM's or TABLE's data definition as the data format for the
+topic. If a message in the topic can't be decoded according to that data format, KSQL considers this message to be
+corrupt. For example, a message is corrupt if KSQL expects message values to be in JSON format, but they are in
+DELIMITED format. The default value in KSQL is ``true``. For example, to ignore corrupt messages, add this to your
+properties file:
+
+.. code:: java
+
+    fail.on.deserialization.error=false
+
+.. _ksql-schema-registry-url:
+
+------------------------
+ksql.schema.registry.url
+------------------------
+
+The Schema Registry URL path to connect KSQL to.
+
+.. _ksql-service-id:
+
+---------------
+ksql.service.id
+---------------
+
+The service ID of the KSQL server. By default, the service ID of KSQL servers is ``default_``. The service ID is also used as
+the prefix for the internal topics created by KSQL. Using the default value ``ksql.service.id``, the KSQL internal topics
+will be prefixed as ``_confluent-ksql-default_`` (e.g. ``_command_topic`` becomes ``_confluent-ksql-default__command_topic``).
+
+.. _ksql.sink.partitions:
+
+--------------------
+ksql.sink.partitions
+--------------------
+
+The default number of partitions for the topics created by KSQL. The default is four.
+
+.. _ksql.sink.replicas:
+
+------------------
+ksql.sink.replicas
+------------------
+
+The default number of replicas for the topics created by KSQL. The default is one.
+
+KSQL Server Settings
+--------------------
+
+These configurations control the general behavior of the KSQL server. These configurations can only be specified via the
+``ksql-server.properties`` file.
+
+.. important:: KSQL server configuration settings take precedence over those set in the KSQL CLI. For example, if a value
+               for ``ksql.streams.replication.factor`` is set in both the KSQL server and KSQL CLI, the KSQL server value is used.
+
+.. _ksql-queries-file:
+
+-----------------
+ksql.queries.file
+-----------------
+
+A file that specifies a predefined set of queries for the KSQL server, KSQL, and its underlying Kafka Streams instances.
```

Comment:
It may be confusing to specify that the queries are for KSQL server, KSQL, and its underlying Kafka Streams instances. I think just saying KSQL server or KSQL should be sufficient.